alibuild_helpers/build.py

```diff
@@ -336,31 +337,46 @@ def doBuild(args, parser):
                   "  git pull --rebase\n",
                   pwd=os.getcwd(), star=star()))
 
+  # Prefetch git heads in parallel
+  debug("Prefetching git heads in parallel:")
+  poolConfig = {"processes": min(max(len(buildOrder), 1), 32)}
+  pool = ThreadPool(**poolConfig)
+  debug("Created thread pool with config: %s" % poolConfig)
+
+  git_commands = {} # store futures
+  for p in buildOrder:
+    if "source" in specs[p]:
+      # Replace source with local one if we are in development mode.
+      if specs[p]["package"] in develPkgs:
+        specs[p]["source"] = join(os.getcwd(), specs[p]["package"])
+
+      cmd = "git ls-remote --heads %s" % specs[p]["source"]
+      debug("Enqueued '%s'" % cmd)
+      git_commands[p] = pool.apply_async(getstatusoutput, (cmd,))
+
+  for p in buildOrder:
+    if p in git_commands:
+      result = git_commands[p].get()
+      dieOnError(result[0], "Unable to fetch from %s" % specs[p]["source"])
```

Comment:
You still call getstatusoutput inside the thread, right? That will create the process which might end up being zombied.